socketio/src/client/raw_client.rs

```diff
@@ -200,9 +236,86 @@ impl RawClient {
         D: Into<Payload>,
     {
         let id = thread_rng().gen_range(0..999);
-        let socket_packet =
-            self.socket
-                .build_packet_for_payload(data.into(), event.into(), &self.nsp, Some(id))?;
+        let socket_packet = InnerSocket::build_packet_for_payloads(
+            vec![data.into()],
+            Some(event.into()),
+            &self.nsp,
+            Some(id),
+            false,
+        )?;
+
+        let ack = Ack {
+            id,
+            time_started: Instant::now(),
+            timeout,
+            callback: Callback::<SocketCallback>::new(callback),
+        };
+
+        // add the ack to the tuple of outstanding acks
+        self.outstanding_acks.write()?.push(ack);
+
+        self.socket.send(socket_packet)?;
+        Ok(())
+    }
+
+    /// Sends a message to the server but `alloc`s an `ack` to check whether the
+    /// server responded in a given time span. This message takes an event, which
+    /// could either be one of the common events like "message" or "error" or a
+    /// custom event like "foo", as well as a data parameter. But be careful,
+    /// in case you send a [`Payload::String`], the string needs to be valid JSON.
+    /// It's even recommended to use a library like serde_json to serialize the data properly.
+    /// It also requires a timeout `Duration` in which the client needs to answer.
+    /// If the ack is acked in the correct time span, the specified callback is
+    /// called. The callback consumes a [`Payload`] which represents the data send
+    /// by the server.
+    ///
+    /// # Example
+    /// ```
+    /// use rust_socketio::{ClientBuilder, Payload, RawClient};
+    /// use serde_json::json;
+    /// use std::time::Duration;
+    /// use std::thread::sleep;
+    ///
+    /// let mut socket = ClientBuilder::new("http://localhost:4200/")
+    ///     .on("foo", |payload: Payload, _| println!("Received: {:#?}", payload))
+    ///     .connect()
+    ///     .expect("connection failed");
+    ///
+    /// let ack_callback = |message: Payload, socket: RawClient| {
+    ///     match message {
+    ///         Payload::String(str) => println!("{}", str),
+    ///         Payload::Binary(bytes) => println!("Received bytes: {:#?}", bytes),
+    ///         Payload::Number(num) => println!("Received number: {}", num),
+    ///    }
+    /// };
+    ///
+    /// let payload = json!({"token": 123});
+    /// socket.emit_with_ack("foo", payload, Duration::from_secs(2), ack_callback).unwrap();
+    ///
+    /// sleep(Duration::from_secs(2));
+    /// ```
+    #[inline]
+    pub fn emit_multi_with_ack<F, E, D>(
```

Comment:
What value does this add? Could we add and `impl From Vec<...> for Payload `, so the existing function works?

Reply:
This function wants to emits `Vec<Payload>` as what javascript version of socketio client did.